core/market/src/negotiation/error.rs

```diff
@@ -134,36 +134,48 @@ pub enum AgreementEventsError {
     Internal(String),
 }
 
-#[derive(Error, Debug)]
+#[derive(Error, Debug, Serialize, Deserialize)]
 pub enum GetProposalError {
     #[error("Proposal [{0}] not found (subscription [{1:?}]).")]
     NotFound(ProposalId, Option<SubscriptionId>),
     #[error("Get Proposal [{0}] (subscription [{1:?}]) internal error: [{2}]")]
     Internal(ProposalId, Option<SubscriptionId>, String),
 }
 
-#[derive(Error, Debug)]
-pub enum ProposalError {
+#[derive(Error, Debug, Serialize, Deserialize)]
+pub enum ProposalValidationError {
     #[error("Subscription [{0}] wasn't found.")]
     NoSubscription(SubscriptionId),
     #[error("Subscription [{0}] was already unsubscribed.")]
     Unsubscribed(SubscriptionId),
     #[error("Subscription [{0}] expired.")]
-    SubscriptionExpired(SubscriptionId),
-    #[error("Can't counter own Proposal [{0}].")]
-    OwnProposal(ProposalId),
+    Expired(SubscriptionId),
```

Comment:
Should be `SubscriptionExpired`, because it is Proposal error, so `Expired` would suggest, that Proposal expired.

Reply:
indeed. I was aware once, and then still decided to shorten the name. Will revert this.